watertap/tools/oli_api/credentials.py

```diff
@@ -238,21 +239,39 @@ def set_access_key(self):
 
     # TODO: improve header management for class
     # TODO: possibly save api keys as objects with expiry and other attributes
-    def generate_oliapi_access_key(self, expiry_timecode):
+    def generate_oliapi_access_key(self, key_lifetime=None):
         """
         Generate an access key for OLI Cloud.
 
-        :param expiry_timecode: unix time stamp in milliseconds
+        :param key_lifetime: integer number of days key will be valid
 
         :return string: Response text containing the access key information or an error message
         """
 
+        def _set_expiry_timestamp(key_lifetime):
+            """
+            Set expiry date for OLI Cloud access key.
+
+            :param key_lifetime: integer number of days key will be valid (up to 365)
+
+            :return unix_timestamp_ms: unix timestamp (in ms) for when key will expire
+            """
+
+            if key_lifetime < 1 or key_lifetime > 365:
```

Comment:
Should you also check `if key_lifetime is None`? I am not sure the check for <1 or >365 will work if the arg is None

Reply:
I think I'll change the default to `365` instead of `None`